extensions/interactions/ItemSelectionInput/directives/oppia-response-item-selection-input.component.ts

```diff
@@ -13,28 +13,47 @@
 // limitations under the License.
 
 /**
- * @fileoverview Directive for the ItemSelectionInput response.
+ * @fileoverview Component for the ItemSelectionInput response.
  *
  * IMPORTANT NOTE: The naming convention for customization args that are passed
  * into the directive is: the name of the parameter, followed by 'With',
  * followed by the name of the arg.
  */
 
-require('services/html-escaper.service.ts');
+import { Component, Input, OnInit } from '@angular/core';
+import { downgradeComponent } from '@angular/upgrade/static';
+import { HtmlEscaperService } from 'services/html-escaper.service';
 
-angular.module('oppia').component('oppiaResponseItemSelectionInput', {
-  template: require('./item-selection-input-response.component.html'),
-  controllerAs: '$ctrl',
-  controller: ['$attrs', 'HtmlEscaperService',
-    function($attrs, HtmlEscaperService) {
-      var ctrl = this;
-      ctrl.$onInit = function() {
-        const choices = HtmlEscaperService.escapedJsonToObj($attrs.choices);
-        const answer = HtmlEscaperService.escapedJsonToObj($attrs.answer);
+@Component({
+  selector: 'oppia-response-item-selection-input',
+  templateUrl: './item-selection-input-response.component.html',
+  styleUrls: []
+})
+export class ResponseItemSelectionInputComponent implements OnInit {
+  // These properties are initialized using Angular lifecycle hooks
+  // and we need to do non-null assertion, for more information see
+  // https://github.com/oppia/oppia/wiki/Guide-on-defining-types#ts-7-1
+  @Input() answer!: string;
+  @Input() choices!: string;
+  responses: string[];
 
-        const choicesContentIds = choices.map(choice => choice._contentId);
-        ctrl.answer = answer.map(
-          contentId => choices[choicesContentIds.indexOf(contentId)]._html);
-      };
-    }]
-});
+  constructor(private htmlEscaperService: HtmlEscaperService) {}
+
+  ngOnInit(): void {
+    const _answer = this.htmlEscaperService.escapedJsonToObj(
```

Comment:
```suggestion
    const answer = this.htmlEscaperService.escapedJsonToObj(
```

`_` prefix indicates that the variable is private which is not the case here.

Here and also below for choices.

Reply:
Done